File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @param {number} n
     * @return {ListNode}
     */
    removeNthFromEnd(head, n) {
        let dummy = new ListNode(0);
        dummy.next = head;
        let stack = [dummy];
        let cur = head;
        while (cur) {
            stack.push(cur);
            cur = cur.next;
        }

        let prev;
        for (let i = 0; i <= n; i++) {
            prev = stack.pop();
        }

        prev.next = prev.next.next;

        return dummy.next;
    }
}

File: Data Structures & Algorithms/reverse-bits/submission-1.java
class Solution {
    public int reverseBits(int n) {
        int res = 0;

        for (int i = 0; i < 32; i++) {
            int cur = ((n >> i) & 1);

            if ((cur & 1) == 1) {
                res = (res | (1 << (31 - i)));
            }
        }
        return res;
    }
}
